src/commands/xmentor.ts

```diff
@@ -1,60 +1,89 @@
-import { GluegunToolbox } from 'gluegun'
+import { GluegunToolbox } from 'gluegun';
+
+const LANGUAGE_TEST_FILE_MAPPER: { [k: string]: string } = {
+  'typescript': 'test.ts',
+  'javascript': 'spec.js',
+};
+
+const LANGUAGE_RUN_COMMAND_MAPPER: { [k: string]: string } = {
+  'typescript': `npx tsc --noEmit -p . && npx jest --i --no-cache`,
+  'javascript': 'jest --no-cache ./*',
+};
 
 export default {
   name: 'xmentor',
   run: async (toolbox: GluegunToolbox) => {
-    const { print, parameters, system, prompt, filesystem } = toolbox
+    const { print, parameters, system, prompt, filesystem } = toolbox;
 
-    const exerciseUUID = parameters.string
+    const exerciseUUID = parameters.string;
+    const spinner = print.spin({ text: 'Downloading student solution.', spinner: 'dots' });
 
-    const spinner = print.spin({ text: 'Downloading student solution.', spinner: 'dots' })
+    if (!exerciseUUID) return spinner.fail('Please provide an UUID');
 
     try {
-      const pathToExercise = (await system.run(`exercism download --uuid=${exerciseUUID}`)).trim()
-      const fileName = pathToExercise.substring(pathToExercise.lastIndexOf('\\') + 1)
-      spinner.succeed()
+      const pathToExercise = (await system.run(`exercism download --uuid=${exerciseUUID}`)).trim();
+      if (!pathToExercise) throw new Error('Path to exercise missing');
```

Comment:
I have a hard time seeing when this would happen unless the download didn't work, in which case exercism CLI would throw an error anyway. Still, there might be an edge case out there that I can't think about and this will take care of it so lets keep it.